src/main/webapp/js/components/csv/csv-config-page.jsx

```diff
@@ -0,0 +1,259 @@
+      displayName: 'Related field',
+      required: true,
+      relatedField,
+      relatedEntity,
+    },
+    {
+      name: 'fieldValue',
+      fieldType: TEXT,
+      displayName: 'Related field CSV name',
+      required: true,
+    },
+  ];
+
+  validate = (values) => {
+    const errors = {};
+    errors.csvFields = [];
+
+    _.forEach(values.csvFields, (csvField, key) => {
+      _.forEach(this.getFields(values.csvFields), (config) => {
+        const val = csvField[config.name];
+
+        if (config.required && !config.hidden && (_.isNil(val) || val === '')) {
+          errors.csvFields[key] = { [config.name]: `${config.displayName} is required` };
+        }
+      });
+    });
+
+    return errors;
+  };
+
+  render() {
+    const { entityType } = this.props;
+
+    return (
+      <div>
+        <Form
+          onSubmit={this.onSubmit}
+          initialValues={this.props.csvConfig}
+          validate={this.validate}
+          mutators={{
+            ...arrayMutators,
+          }}
+          render={({
+            handleSubmit, values, invalid,
+          }) => (
+            <form onSubmit={handleSubmit}>
+              { !_.isEmpty(this.props.csvConfig) ? (
+                <div className="input-row required">
+                  <span className="col-md-2 col-form-label"> CSV Name </span>
+                  <span style={{ fontSize: '20px' }}>
+                    {' '}
+                    {this.props.csvConfig.name}
+                    {' '}
+                  </span>
+                </div>
+              ) : (
+                <TextField
+                  key="name"
+                  placeholder="Name"
+                  fieldConfig={{
+                    name: 'name', displayName: 'CSV Name', required: true,
+                  }}
+                />
+              )}
+              <FieldArray name="csvFields">
+                { ({ fields }) => (
+                  <div>
+                    {fields.map((name, index) => (
+                      <div key={name} className="csvFields">
+                        {_.map(this.getFields(fields.value[index]), elem => renderFormField({ ...elem, name: `${name}.${elem.name}` }))}
+                        {(() => {
+                          // eslint-disable-next-line max-len
+                          const fieldConfig = this.props.fieldConfig[fields.value[index].fieldConfigId];
+                          if (fieldConfig && fieldConfig.fieldType === RELATION) {
+                            return (
+                              <div>
+                                <Field name="fieldValueMap" component="">
+                                  <FieldArray name={`${name}`}>
+                                    { ({ fields: relatedFields }) => (
+                                      <div>
+                                        {relatedFields.map(relatedName => (
+                                          <div key={relatedName} className="csvRelatedField">
```

Comment:
rename all of those 'relatedName' the class name 'csvRelatedField' and the map variable 'relatedElem'